modules/netIdSystem.js

```diff
@@ -18,22 +23,108 @@ export const netIdSubmodule = {
+      utils.logInfo(LOG_PREFIX + 'id already in config', config.value);
+      return { id: config.value };
+    }
+    if (storedId) {
+      utils.logInfo(LOG_PREFIX + 'id already in cache', storedId);
+      return { id: storedId };
+    }
+    const url = `https://einwilligungsspeicher.netid.de/netid-subject-identifiers?q.tapp_id.eq=${config.params.tapp_id}`;
+    const resp = function (callback) {
+      const callbacks = {
+        success: response => {
+          let responseObj;
+          let responseRaw;
+          if (response) {
+            try {
+              responseRaw = JSON.parse(response);
+              utils.logInfo(LOG_PREFIX + 'response received from the server', responseRaw);
+              if (responseRaw.hasOwnProperty('tpid') && responseRaw.hasOwnProperty('status_code') && responseRaw.status_code == 'OK') {
+                responseObj = { 'netId': responseRaw.tpid };
+                utils.logInfo(LOG_PREFIX + 'used tpid', responseObj);
+              }
+            } catch (error) {
+              utils.logError(LOG_PREFIX + error);
+            }
+          }
+          callback(responseObj);
+        },
+        error: error => {
+          let responseObj;
+          utils.logError(LOG_PREFIX + 'getId fetch encountered an error', error);
+          responseObj = { 'netId': null };
+          callback(responseObj);
+        }
+      };
+      utils.logInfo(LOG_PREFIX + 'requesting an ID from the server');
+      ajax(url, callbacks, undefined, { method: 'GET', withCredentials: true });
+    };
+    return { id: storedId, callback: resp };
+  }
+};
+function hasRequiredConfig(config) {
+  if (!config || !config.params || !config.params.tapp_id || utils.isEmptyStr(config.params.tapp_id)) {
+    utils.logError(LOG_PREFIX + 'tapp_id is undefined or empty');
+    return false;
+  }
+
+  if (!config.storage || !config.storage.type || !config.storage.name) {
+    utils.logError(LOG_PREFIX + 'storage required to be set');
+    return false;
+  }
+
+  return true;
+};
+function checkAbTestingConfig(config, userId) {
+  if (config && config.params && config.params.abTesting && config.params.abTesting.enabled && config.params.abTesting.controlGroupPct) {
+    if (!utils.isNumber(config.params.abTesting.controlGroupPct) || config.params.abTesting.controlGroupPct < 0 || config.params.abTesting.controlGroupPct > 1) {
+      utils.logError(LOG_PREFIX + 'A/B Testing controlGroupPct must be a number >= 0 and <= 1! Skipping A/B Testing');
+      return { enabled: false };
+    } else {
+      return { enabled: true, controlGroup: abTestBucket(userId) < config.params.abTesting.controlGroupPct * ABTEST_RESOLUTION }
+    }
+  }
+  return { enabled: false };
+};
+function abTestBucket(userId) {
+  if (userId) {
+    return ((utils.cyrb53Hash(userId) % ABTEST_RESOLUTION) + ABTEST_RESOLUTION) % ABTEST_RESOLUTION;
+  } else {
+    return Math.floor(Math.random() * ABTEST_RESOLUTION);
```

Comment:
Will this condition be hit if prior to this function execution, within the decode function, you are returning undefined if 'netid' property of value type is not a string. Also, that check on line 31 should check for the existence of 'netid' on the value object as well.